acceptance-tests/s3_test.go

```diff
@@ -45,4 +46,58 @@ var _ = Describe("S3", Label("s3"), func() {
 		By("deleting the file from bucket using the second app")
 		appTwo.DELETE(filename)
 	})
+
+	It("can be accessed only through HTTPS", func() {
+		By("creating a service instance")
+		serviceInstance := services.CreateInstance(
+			"csb-aws-s3-bucket",
+			services.WithPlan("default"),
+			services.WithParameters(map[string]any{
+				"require_tls":          true,
+				"acl":                  "public-read",
+				"boc_object_ownership": "ObjectWriter",
+			}),
+		)
+		defer serviceInstance.Delete()
+
+		By("pushing the unstarted app")
+		app := apps.Push(apps.WithApp(apps.S3))
+		defer apps.Delete(app)
+
+		By("binding the app to the s3 service instance")
+		binding := serviceInstance.Bind(app)
+		defer binding.Unbind()
+
+		By("starting the app")
+		apps.Start(app)
+
+		By("uploading a file using the app")
+		filename := random.Hexadecimal()
+		fileContent := fmt.Sprintf("This is a dummy file that will be uploaded the S3 at %s.", time.Now().String())
+		app.PUT(fileContent, "/upload-with-public-read-acl/%s", filename)
+
+		By("downloading the file using the app and the AWS Go library")
+		got := app.GET(filename).String()
+		Expect(got).To(Equal(fileContent))
+
+		By("downloading the file using the app and HTTPS request")
+		got = app.GET("/check-https/%s", filename).String()
+		Expect(got).To(Equal(fileContent))
+
+		By("receiving a failure when using the app and HTTP request")
+		resp := app.GETResponse("/check-http/%s", filename)
+		defer resp.Body.Close()
+		Expect(resp.StatusCode).To(BeNumerically("==", http.StatusForbidden))
+
+		By("updating the service instance to admit HTTP requests")
+		serviceInstance.Update(services.WithParameters(map[string]any{"require_tls": false}))
+
+		By("receiving a successful response when using the app and HTTP request")
+		resp = app.GETResponse("/check-http/%s", filename)
+		defer resp.Body.Close()
+		Expect(resp.StatusCode).To(BeNumerically("==", http.StatusOK))
+
+		By("deleting the file from bucket using the app")
+		app.DELETE(filename)
```

Comment:
Just to validate my understanding:

This operation is not strictly needed, is it?
My guess is that `defer serviceInstance.Delete()`:
1. Initiates a delete-service operation, which in turn
1. Initiates a terraform destroy, which in turn
1. Initiates a bucket deletion, which in turn
1. Deletes all "non-locked" objects in it

Again, I'm not suggesting to remove this line. Just checking whether my understanding is correct. :)

Reply:
Thank you very much @fnaranjo-vmw!! I would say a bucket won't be deleted if it's not empty